### PR TITLE
Add API ML sample extension package

### PR DESCRIPTION
**Pull request by taban03**
Signed-off-by: at670475 <andrea.tabone@broadcom.com>

Please check if your PR fulfills the following requirements. This is simply a reminder of what we are going to look for before merging your PR. If you don't know all of this information when you create this PR, don't worry. You can edit this template as you're working on it.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [ ] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [ ] Feature 
- [ ] Other... Please describe:

#### Relevant issues

Fixes 

#### Changes proposed in this PR

- 

#### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

#### Does this PR do something the person installing Zowe should know about?

---
* Affected function: _general area of interest_ *

---
* Description: _1 line description_ *

---
* Part: _name of customizable file involved_  *

---
_multi-line description_

#### Is there a related doc issue or Pull Request? 

Doc issue/PR number:

### CODE REVIEW

**Review by jackjia-ibm — COMMENTED** [review]
We may need to modify here https://github.com/zowe/zowe-install-packaging/blob/299831bb9207a4648aff6ab44e291659fee11ac8/install/zowe-install.sh#L187-L191 to install the new package into `<runtime>/components/` directory.

**Comment by jackjia-ibm** [comment]
Also https://github.com/zowe/zowe-install-packaging/blob/299831bb9207a4648aff6ab44e291659fee11ac8/bin/internal/prepare-environment.sh#L128 to make sure this new component will be started (configured) during startup.

File: tests/sanity/test/apiml/test-api-gateway-sample-extension-controller.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        timeout: 30000,
+      });
+      username = process.env.SSH_USER;
+      password = process.env.SSH_PASSWD;
+      debug(`Explorer server URL: https://${process.env.ZOWE_EXTERNAL_HOST}:${process.env.ZOWE_API_MEDIATION_GATEWAY_HTTP_PORT}`);
+    });
+
+    it('should return the greeting message from the gateway sample extension controller', function() {
+      debug('Verify access to greeting endpoint via /api/v1/greeting');
+      getAndVerifyGreeting('/api/v1/greeting');
```
Comment (jackjia-ibm) [inline tests/sanity/test/apiml/test-api-gateway-sample-extension-controller.js:42]: We may need a `return ` here to tell mocha this test returns a Promise, which is an async test.

Alternative way is define the function as `async function()`, then use `await getAndVerifyGreeting()`.

Comment (taban03) [reply tests/sanity/test/apiml/test-api-gateway-sample-extension-controller.js]: Yes good catch, fixed! Thanks Jack!

**Review by jackjia-ibm — APPROVED** [review]
LGTM, thanks a lot Andrea!

**Review by jackjia-ibm — APPROVED** [review]

**Pull request merged**